suzieq/poller/controller/utils/inventory_models.py

```diff
@@ -36,11 +36,13 @@ class DeviceModel(BaseModel):
 
     @validator('jump_host')
     def jump_host_validator(cls, value):
-        jump_host = value[2:] if value.startswith('//') else value
-        pattern = '([a-zA-Z0-9-.]{1,253})@([a-zA-Z0-9-.]{1,253})'
-        assert re.fullmatch(pattern, jump_host), \
-               'format username@jumphost required'
-        return value
+        '''Validate jump host format'''
+        try:
+            uri = parse_url(value)
+            assert uri.scheme is None, \
```

Comment:
This validation approach is too broad. For example a value like `ssh://user@` passes the validation since `uri.scheme` is `ssh`. We could just fix the regex to optionally accept the port:
```
pattern = '([a-zA-Z0-9-.]{1,253})@([a-zA-Z0-9-.]{1,253})(\:\d+)?'
```
See this [example](https://regex101.com/r/MCnAGc/1)

Reply:
It won't pass because I assert the scheme is None.

Reply:
```
In [2]: uri = parse_url('ssh://128.0.1.1')

In [3]: assert uri.scheme is None
---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
<ipython-input-3-ff16f5e513e8> in <module>
----> 1 assert uri.scheme is None

AssertionError: 
```

Reply:
I did what you suggested. It doesn't matter which one we pick. Doing what you said stayed closest to the original and is the strictest.

Reply:
Actually, I take it back. The pattern provided doesn't support an IPv6 host. I don't want to reopen this thread when someone uses ipv6. I've gone back to using parse_url.

Reply:
```
➜ python suzieq/poller/sq_poller.py --syntax-check -I /tmp/inv.yml
2022-06-03 21:31:48,970 - suzieq.poller.controller - WARNING - log level WARNING
2022-06-03 21:31:49,051 - suzieq.poller.controller - ERROR - 
- devices.proximity:
  + jump-host: format username@jumphost[:port] required


suzieq on  fix-737 [$!?] is 📦 v0.18.0a5 via 🐍 v3.8.13 (.venv) 
❯ cat /tmp/inv.yml 
sources:
  - name: LD8-PC-BROA
    hosts:
      - url: ssh://10.1.1.6:22 devtype=eos
      - url: ssh://10.1.1.5:22 devtype=eos
      - url: ssh://10.1.1.11:22 devtype=nxos
      - url: ssh://10.1.1.12:22 devtype=nxos
auths:
  - name: network-ro
    username: ####
    password: ####


devices:
  - name: proximity
    port: 2222
    ignore-known-hosts: true
    jump-host: ssh://ser@172.21.20.2:212
    jump-host-key-file: id_ed25519
```

Reply:
```
➜ python suzieq/poller/sq_poller.py --syntax-check -I /tmp/inv.yml
2022-06-03 21:32:57,029 - suzieq.poller.controller - WARNING - log level WARNING
Inventory syntax check passed
suzieq on  fix-737 [$!?] is 📦 v0.18.0a5 via 🐍 v3.8.13 (.venv) 
➜ cat /tmp/inv.yml
sources:
  - name: LD8-PC-BROA
    hosts:
      - url: ssh://10.1.1.6:22 devtype=eos
      - url: ssh://10.1.1.5:22 devtype=eos
      - url: ssh://10.1.1.11:22 devtype=nxos
      - url: ssh://10.1.1.12:22 devtype=nxos
auths:
  - name: network-ro
    username: ####
    password: ####


devices:
  - name: proximity
    port: 2222
    ignore-known-hosts: true
    jump-host: user@[2000:1::1]:212
```